examples/utils/migration-tools/README.md

```diff
@@ -2,7 +2,93 @@
 
-See [GitHub](https://github.com/microsoft/FluidFramework) for more details on the Fluid Framework and packages within.
+Use of the migration tools imposes several requirements on the container code and application, detailed here.
+
+## Implementing `IMigratableModel`
+
+Your data model must implement `IMigratableModel` to be migrated using the migration tools.
+
+This includes:
+1. A `version` string to identify the model version.
+1. Methods to export and import data, and to detect if the model supports a given data format:
+	1. `importData: (initialData: ImportType) => Promise<void>`
+	1. `exportData: () => Promise<ExportType>`
+	1. `supportsDataFormat: (initialData: unknown) => initialData is ImportType`
+1. A `dispose` method to clean up the container - most likely calling `IContainer.dispose`.
+
+## Implementing the composite runtime pattern
+
+See documentation for the composite runtime pattern [here](./src/compositeRuntime/README.md).
+
+The migration tools expect to find an `IMigratableModel` by accessing and calling a `getModel()` function provided on the `entryPoint`.  They also expect to find an `IMigrationTool` by accessing a `migrationTool` member of the `entryPoint`.  These requirements are most easily satisfied by using the composite runtime pattern.
+
+`getModel()` is a function that takes an `IContainer` to aid in producing the `IMigratableModel`.  This is because the contract of `IMigratableModel` likely requires functionality from `IContainer` (especially `IContainer.dispose()`).
+
+### Defining the entry point piece
+
+```ts
+const rootDatastoreAlias = "my-root-datastore";
+
+export const getModelEntryPointPiece: IEntryPointPiece = {
+	name: "getModel",
+	registryEntries: [MyRootDatastoreFactory.registryEntry],
+	onCreate: async (runtime: IContainerRuntime): Promise<void> => {
+		const rootDatastore = await runtime.createDataStore(MyRootDatastoreFactory.type);
+		await rootDatastore.trySetAlias(rootDatastoreAlias);
+	},
+	onLoad: async (runtime: IContainerRuntime): Promise<void> => {},
+	createPiece: async (runtime: IContainerRuntime): Promise<MyRootDatastore> => {
+		const entryPointHandle = await containerRuntime.getAliasedDataStoreEntryPoint(rootDatastoreAlias);
+
+		if (entryPointHandle === undefined) {
+			throw new Error(`Default dataStore [${rootDatastoreAlias}] must exist`);
+		}
+
+		// Entry points are typed as FluidObject and must be cast.  Here we know it's a MyRootDatastore since
+		// we created it just above.  Type validation can be added here if desired.
+		const rootDatastore = entryPointHandle.get() as Promise<MyRootDatastore>;
+		// MigratableAppModel (defined by the container code author) must implement IMigratableAppModel.
+		// Note that we're returning a function of type (container: IContainer) => Promise<IMigratableModel>
+		return async (container: IContainer) => new MigratableAppModel(rootDatastore, container);
+	},
+};
+```
```

Comment:
If I understand this correctly, I think it would be helpful to know what the original `IEntryPointPiece` was or how `instantiateRuntime` was defined. I think this replaces the example `rootDatastoreEntryPointPiece` shown in the `compositeRuntime` readme. (It would be a mistake to have that original piece and this new `getModel` piece in the same runtime. (yeah?)

Reply:
Like show the delta of before/after adding migration support?  I might take that as a followup, because adding migration support to an existing implementation requires a few extra steps to ensure the migration tool is instantiated post-creation as well.

This draft would only explain how to create your brand new code with future migration in mind, but doesn't cover how to retroactively add migration support to existing containers.  Though I agree that is a top scenario to explain and will be doing that ultimately.

> It would be a mistake to have that original piece and this new getModel piece in the same runtime. (yeah?)

Not necessarily a mistake, might depend on whether the author wanted to retain a simple access to the root datastore without the `IMigratableModel` superpowers maybe.  You'd probably want to be thoughtful about which `IEntryPointPiece` does the `onCreate` steps, though theoretically the aliasing feature would handle any racing.  Agree that I should probably figure out best practices to recommend here though.